llvm/lib/Analysis/StackSafetyAnalysis.cpp

```diff
@@ -273,6 +273,17 @@ ConstantRange StackSafetyLocalAnalysis::offsetFrom(Value *Addr, Value *Base) {
     return UnknownRange;
 
   auto *PtrTy = PointerType::getUnqual(SE.getContext());
```

Comment:
Didn't follow the context, but PointerType::getUnqual should never be used. Can this just use Addr->getType()?

Reply:
I also found another use of `getUnqual` in this pass that still causes the assert to trip, so indeed seems like we should get rid of those instead. I'll try it

Reply:
I tried it and the pass messes up somewhere else instead. It seems like the pass isn't written with handling address spaces with differently sized pointers at all. It even hardcodes the pointer size for AS 0 and uses it in a bunch of places.

The changes needed to the pass seems to be big, I don't mind looking at it sometime but this is really a blocker so we need a quick bandaid first.

Reply:
Based on the context, I'm guessing another quick bandaid would be to replace the hardcoded PtrTy with the alloc address space pointer type

Reply:
I'd expect any fix to this to at least remove all the getTruncateOrZeroExtend() calls on pointer SCEVS in this pass, as that fundamentally never makes sense.

Reply:
I updated the diff so `getTruncateOrZeroExtend` is no longer called for pointers